### PR TITLE
feat: add pricing connections

### PR DESCRIPTION
**Pull request by ServiceDog**

### CODE REVIEW

File: openapi/types.rs [hunk 1]
```diff
@@ -82,7 +82,16 @@ pub struct FlightOption {
...
+    pub timestamp_arrive: SystemTime,
+
+    /// The estimated trip distance
+    pub distance_km: f32,
+    
+    /// The currency type, e.g. USD, EUR
+    pub currency_type: Option<String>,
+
+    /// The cost of the trip for the customer
+    pub customer_cost: Option<f32>
```
Comment (GoodluckH) [inline openapi/types.rs:94]: Maybe call this `base_pricing`? This is the upfront cost calculated by `svc-pricing`. And if we do layer in auction later on, we would have an additional field called `final_pricing` or `accepted_pricing`.

File: server/src/main.rs [hunk 2]
```diff
@@ -147,6 +148,85 @@ pub async fn query_vertiports(
...
 
+/// Requests the cost of an itinerary from svc-pricing
+///
+/// # Arguments
+/// * _flight - The flight plan to obtain the cost of
+/// * client  - The active client for the svc-pricing server
+///
+/// # Returns
+/// A tuple of price and currency type
+async fn customer_cost(
```
Comment (GoodluckH) [inline server/src/main.rs:159]: Maybe rename to `get_base_pricing`?

It's a good practice to start with a verb for a function.

**Review by GoodluckH — CHANGES_REQUESTED** [review]

File: server/Cargo.toml [hunk 3]
```diff
@@ -10,27 +10,26 @@ edition = "2021"            # Allowed values are 2015, 2018, 2021
 vendored-openssl = ['openssl/vendored']
 
 [dependencies]
-axum                    = "0.5.15"
-cargo-husky             = "1"
-chrono                  = { version = "0.4.22", features = ["serde"] }
-futures                 = "0.3"
-hyper                   = "0.14.20"
-openssl                 = "0.10"
```
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/Cargo.toml:18]: openssl was added back in with a force push, curious that it still shows as being dropped

File: server/src/main.rs (hunk 2 above)
Comment (ServiceDog) [reply server/src/main.rs:159]: It's actually idiomatic rust to not use "get" for getters, but to use "set" for setters:

https://github.com/rust-lang/rfcs/blob/master/text/0344-conventions-galore.md#gettersetter-apis

Will change to "base_pricing"

File: server/src/main.rs [hunk 4]
```diff
@@ -147,6 +148,85 @@ pub async fn query_vertiports(
...
+}
+
+/// Scrapes the incoming flight plans for information the customer wants
+///
+/// # Arguments
+/// * _flight - The flight plan to obtain the cost of
+/// * client  - The active client for the svc-pricing server
+///
+/// # Returns
+/// A tuple of price and currency type
```
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/main.rs:189]: Copy pasta error

File: server/src/main.rs [hunk 5]
```diff
@@ -147,6 +148,85 @@ pub async fn query_vertiports(
...
+        return None;
+    };
+
+    Some(rest_types::FlightOption {
+        fp_id: plan.id.clone(),
+        vertiport_depart_id: plan.vertiport_depart_id.to_string(),
+        vertiport_arrive_id: plan.vertiport_arrive_id.to_string(),
+        timestamp_depart: time_depart,
+        timestamp_arrive: time_arrive,
+        distance_km: (plan.estimated_distance as f32) / 1000.0,
```
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/main.rs:224]: I agree, this field is defined in the svc-scheduler proto file: https://github.com/Arrow-air/svc-scheduler/blob/develop/proto/svc-scheduler-grpc.proto#L90 as a U32 representing distance in meters

I wouldn't mind switching to meters, I think in either case it should be an F32

File: server/src/main.rs [hunk 6]
```diff
@@ -225,45 +299,67 @@ pub async fn query_flight(
...
+    let pricing_option = grpc_clients.pricing.get_client().await;
+    if let Some(mut pricing_client) = pricing_option {
+        for mut x in &mut flights {
+            let (cost, currency) = customer_cost(x, &mut pricing_client).await;
+            x.customer_cost = cost;
+            x.currency_type = currency;
         }
-        Err(e) => Err((StatusCode::CONFLICT, e.to_string())),
+    } else {
+        // Allow flights to go back to customer without pricing?
```
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/main.rs:353]: Was thinking about your use case where they get a flight option back and it says "we're creating an auction for this flight"

File: server/src/main.rs [hunk 7]
```diff
@@ -147,6 +148,85 @@ pub async fn query_vertiports(
...
+            (None, None)
+        }
+    }
+}
+
+/// Scrapes the incoming flight plans for information the customer wants
+///
+/// # Arguments
+/// * _flight - The flight plan to obtain the cost of
+/// * client  - The active client for the svc-pricing server
```
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/main.rs]: https://github.com/Arrow-air/svc-cargo/pull/14/commits/5bfa8c58e96525ccf1913bdf1e3ff75f85a5c8e8

File: server/src/main.rs [hunk 8]
```diff
@@ -147,6 +148,85 @@ pub async fn query_vertiports(
     }
 }
 
+/// Requests the cost of an itinerary from svc-pricing
+///
+/// # Arguments
+/// * _flight - The flight plan to obtain the cost of
```
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/main.rs]: https://github.com/Arrow-air/svc-cargo/pull/14/commits/5bfa8c58e96525ccf1913bdf1e3ff75f85a5c8e8

File: server/src/main.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/main.rs]: https://github.com/Arrow-air/svc-cargo/pull/14/commits/5bfa8c58e96525ccf1913bdf1e3ff75f85a5c8e8

File: server/src/main.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/main.rs]: https://github.com/Arrow-air/svc-cargo/pull/14/commits/5bfa8c58e96525ccf1913bdf1e3ff75f85a5c8e8

File: openapi/types.rs (hunk 1 above)
Comment (ServiceDog) [reply openapi/types.rs]: https://github.com/Arrow-air/svc-cargo/pull/14/commits/5bfa8c58e96525ccf1913bdf1e3ff75f85a5c8e8

File: server/src/main.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (GoodluckH) [reply server/src/main.rs:352]: Yea that's why the pricing computed by `svc-pricing` live in the `base_pricing` field, which for now intended to be an immutable field. Auction pricing will be a mutable field, to be named, which updates as bidders submit their bids. Then, we'd probably have a `confirmed_pricing` field, which is the final price that's accepted by both the shipper and the operator. This is the price that has contractural effect and will be used by `svc-payment`.

For an auction, there has to be somewhere to start the auction from. So, in either cases, we can't have a null pricing value.

File: server/src/main.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (romanmandryk) [reply server/src/main.rs]: In scheduler I changed distance to meters as it was in U32 meters in storage. The issue with floats is they can't implement `Eq` and `Ord` traits and that causes some headaches with clippy, generated rust code from grpc and in general with ordering. But in the end floats work too, just with more trouble.

[the remark this answers is not in the document]
Comment (GoodluckH) [reply server/src/main.rs]: In router lib, this was handled by ordered-float crate.

File: server/src/main.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/main.rs]: https://github.com/Arrow-air/svc-cargo/pull/14/commits/2fc70b199aff09625dc964bc40b98fa5d2c9a31f

**Review by GoodluckH — APPROVED** [review]

**Pull request merged**